sdk/src/main/java/com/mapbox/maps/renderer/gl/TextureRenderer.kt

```diff
@@ -0,0 +1,156 @@
+        0f, 0f,
+        1f, 0f,
+        0f, 1f,
+        1f, 1f,
+      )
+    )
+
+    val vertexShader = GlUtils.loadShader(
+      GLES20.GL_VERTEX_SHADER,
+      VERTEX_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    val fragmentShader = GlUtils.loadShader(
+      GLES20.GL_FRAGMENT_SHADER,
+      FRAGMENT_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    program = GLES20.glCreateProgram().also {
+      GlUtils.checkError("glCreateProgram")
+      GLES20.glAttachShader(it, vertexShader)
+      GlUtils.checkError("glAttachShader")
+
+      GLES20.glAttachShader(it, fragmentShader)
+      GlUtils.checkError("glAttachShader")
+
+      GLES20.glLinkProgram(it)
+      GlUtils.checkError("glLinkProgram")
+    }
+    attributePosition = GLES20.glGetAttribLocation(program, "aPosition")
+    attributeTextureCoord = GLES20.glGetAttribLocation(program, "aTexCoord")
+
+    uniformTexture = GLES20.glGetUniformLocation(program, "uTexture")
+  }
+
+  fun render(textureID: Int) {
+    if (program == 0) {
+      initialize()
+    }
+    if (program != 0) {
+      GLES20.glUseProgram(program)
+      GLES20.glBindBuffer(GLES20.GL_ARRAY_BUFFER, vbo[0])
+      GLES20.glVertexAttribPointer(
+        attributePosition,
+        COORDS_PER_VERTEX,
+        GLES20.GL_FLOAT,
+        false,
+        VERTEX_STRIDE,
+        0
+      )
+      GLES20.glEnableVertexAttribArray(attributePosition)
+      GLES20.glBindBuffer(GLES20.GL_ARRAY_BUFFER, vbo[1])
+      GLES20.glVertexAttribPointer(
+        attributeTextureCoord,
+        COORDS_PER_TEX,
+        GLES20.GL_FLOAT,
+        false,
+        TEX_STRIDE,
+        0
+      )
+      GLES20.glEnableVertexAttribArray(attributeTextureCoord)
+
+      GLES20.glActiveTexture(GLES20.GL_TEXTURE0)
+      GLES20.glBindTexture(GLES20.GL_TEXTURE_2D, textureID)
+      GLES20.glUniform1i(uniformTexture, 0)
+
+      GLES20.glDrawArrays(GLES20.GL_TRIANGLE_STRIP, 0, VERTEX_COUNT)
+
+      GLES20.glDisableVertexAttribArray(attributePosition)
+      GLES20.glDisableVertexAttribArray(attributeTextureCoord)
+      GLES20.glBindBuffer(GLES20.GL_ARRAY_BUFFER, 0)
+      GLES20.glUseProgram(0)
+    }
+  }
+
+  fun deinitialize() {
+    if (program != 0) {
+      GLES20.glDeleteBuffers(vbo.size, vbo, 0)
+      GLES20.glDeleteProgram(program)
+      program = 0
+    }
+  }
+
+  private fun setupVbo(vertexArray: FloatArray, textureArray: FloatArray) {
+    GLES20.glGenBuffers(2, vbo, 0)
+    GLES20.glBindBuffer(GLES20.GL_ARRAY_BUFFER, vbo[0])
+    GLES20.glBufferData(
+      GLES20.GL_ARRAY_BUFFER,
+      vertexArray.size * 4,
+      vertexArray.toFloatBuffer(),
+      GLES20.GL_STATIC_DRAW
+    )
+    GLES20.glBindBuffer(GLES20.GL_ARRAY_BUFFER, vbo[1])
+    GLES20.glBufferData(
+      GLES20.GL_ARRAY_BUFFER,
+      textureArray.size * 4,
+      textureArray.toFloatBuffer(),
+      GLES20.GL_STATIC_DRAW
+    )
+    GLES20.glBindBuffer(GLES20.GL_ARRAY_BUFFER, 0)
+  }
+
+  companion object {
```

Comment:
could be private here instead of all vars below

Reply:
Yep